File: week3/AhmetTahaSener/BinarySearch.cs
public class Solution
{
    public int Search(int[] nums, int target)
    {
        int min = 0;
        int max = nums.Length - 1;
        int mid = 0;
        while (min <= max)
        {
            mid = (min + max) / 2;
            if (target == nums[mid])
            {
                return mid;
            }
            else if (target > nums[mid])
            {
                min = mid + 1;
            }
            else if (target < nums[mid])
            {
                max = mid - 1;
            }
        }
        return -1;
    }
}

File: week3/Erenimo3442/binary_search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low = 0
        high = len(nums) - 1
        med = int((low + high) / 2)
        while True:
            med = int((low + high) / 2)
            if nums[med] > target:
                high = med - 1
                med = int((low + high) / 2)
            elif nums[med] < target:
                low = med + 1
                med = int((low + high) / 2)
            if nums[med] == target:
                return med
            if low == high or low > high: 
                return -1     